DS/Assignment1/GeneralizedList.java

```diff
@@ -0,0 +1,201 @@
+		while (inputList.charAt(index) != ')' && index < inputList.length()) {
+			if (inputList.charAt(index) == ',') {
+				index++;
+			} else if (inputList.charAt(index) == '(') {
+				index = index + 1;
+				linked.add(createSubList(inputList));
+				index = index + 1;
+			} else {
+				while (inputList.charAt(index) != ','
+						&& inputList.charAt(index) != ')') {
+					digit = digit + inputList.charAt(index);
+					index++;
+				}
+				linked.add(Integer.parseInt(digit));
+				digit = "";
+			}
+		}
+		return linked;
+	}
+
+	@Override
+	public String toString() {
+
+		return "Generalized List is :" + linkedList;
+	}
+
+	// To calculate sum of list
+	public int calculateSum() {
+		int sum = 0;
+		// iterating list
+		Iterator<Object> iterator = linkedList.iterator();
+		while (iterator.hasNext()) {
+			Object obj = iterator.next();
+			// if it is sub linked list
+			if (obj instanceof LinkedList) {
+				sum = sum + calculateSubListSum(obj);
+			} else {// if not list
+				sum = sum + (Integer) obj;
+
+			}
+		}
+		return sum;
+
+	}
+
+	// Calculating sum of sublist
+	private int calculateSubListSum(Object linked) {
+		int sum = 0;
+		// iterating list
+		Iterator<Object> iterator = ((List<Object>) linked).iterator();
+		while (iterator.hasNext()) {
+			Object obj = iterator.next();
+			// if it is sub linked list
+			if (obj instanceof LinkedList) {
+				sum = sum + calculateSubListSum(obj);
+			} else {// if not list
+				sum = sum + (Integer) obj;
+
+			}
+		}
+		return sum;
+	}
+
+	// Function calculating maximum element of the list
+	public int calculateMax() {
+		// iterating list
+		Iterator<Object> iterator = linkedList.iterator();
+		int max = (Integer) linkedList.get(0);
+		while (iterator.hasNext()) {
+			Object obj = iterator.next();
+			// if it is sub linked list
+			if (obj instanceof LinkedList) {
+				int max1 = calculateSubListMax(obj);
+				if (max < max1) {
+					max = max1;
+				}
+
+			} else {// if not list
+				if (max < (Integer) obj) {
+					max = (Integer) obj;
+				}
+			}
+		}
+		return max;
+
+	}
+
+	// Function calculating maximum from a sublist
+	private int calculateSubListMax(Object linked) {
+		// iterating list
+		Iterator<Object> iterator = ((List<Object>) linked).iterator();
+		int max = (Integer) ((List<Object>) linked).get(0);
+		while (iterator.hasNext()) {
+			Object obj = iterator.next();
+			// if it is sub linked list
+			if (obj instanceof LinkedList) {
+				int max1 = calculateSubListMax(obj);
+				if (max < max1) {
+					max = max1;
+				}
+
+			} else {// if not list
+				if (max < (Integer) obj) {
+					max = (Integer) obj;
+				}
+			}
+		}
+		return max;
+	}
+
+	// Function to search an item
+	public boolean searchValue(int value) {
+		// iterating list
+		Iterator<Object> iterator = linkedList.iterator();
+		while (iterator.hasNext()) {
+			Object obj = iterator.next();
+			// if it is sub linked list
+			if (obj instanceof LinkedList) {
+				if (searchSubForValue(obj, value) == true) {
```

Comment:
ACTION: Can we not recursively call searchValue() method here instead creating new method searchSubForValue()?